src/inert.js

```diff
@@ -18,638 +18,322 @@
-   */
-  _unmanageNode(node) {
-    const inertNode = this._inertManager.deregister(node, this);
-    if (inertNode)
-      this._managedNodes.delete(inertNode);
-  }
-
-  /**
-   * Unregister the entire subtree starting at `startNode`.
-   * @param {Node} startNode
-   */
-  _unmanageSubtree(startNode) {
-    composedTreeWalk(startNode, (node) => this._unmanageNode(node));
-  }
-
-  /**
-   * If a descendant node is found with an `inert` attribute, adopt its managed nodes.
-   * @param {Node} node
-   */
-  _adoptInertRoot(node) {
-    let inertSubroot = this._inertManager.getInertRoot(node);
 
-    // During initialisation this inert root may not have been registered yet,
-    // so register it now if need be.
-    if (!inertSubroot) {
-      this._inertManager.setInert(node, true);
-      inertSubroot = this._inertManager.getInertRoot(node);
-    }
-
-    for (let savedInertNode of inertSubroot.managedNodes)
-      this._manageNode(savedInertNode.node);
-  }
-
-  /**
-   * Callback used when mutation observer detects subtree additions, removals, or attribute changes.
-   * @param {MutationRecord} records
-   * @param {MutationObserver} self
-   */
-  _onMutation(records, self) {
-    for (let record of records) {
-      const target = record.target;
-      if (record.type === 'childList') {
-        // Manage added nodes
-        for (let node of Array.from(record.addedNodes))
-          this._makeSubtreeUnfocusable(node);
-
-        // Un-manage removed nodes
-        for (let node of Array.from(record.removedNodes))
-          this._unmanageSubtree(node);
-      } else if (record.type === 'attributes') {
-        if (record.attributeName === 'tabindex') {
-          // Re-initialise inert node if tabindex changes
-          this._manageNode(target);
-        } else if (target !== this._rootElement &&
-                   record.attributeName === 'inert' &&
-                   target.hasAttribute('inert')) {
-          // If a new inert root is added, adopt its managed nodes and make sure it knows about the
-          // already managed nodes from this inert subroot.
-          this._adoptInertRoot(target);
-          const inertSubroot = this._inertManager.getInertRoot(target);
-          for (let managedNode of this._managedNodes) {
-            if (target.contains(managedNode.node))
-              inertSubroot._manageNode(managedNode.node);
-          }
+      if (!nativeShadowDOM) return;
+      // If element doesn't accept shadowRoot, check if it is a potential custom element
+      // https://html.spec.whatwg.org/multipage/scripting.html#valid-custom-element-name
+      if (!matches(rootElement, acceptsShadowRootSelector)) {
+        const potentialCustomElement = rootElement.tagName.indexOf('-') !== -1;
+        if (!potentialCustomElement) return;
+      }
+      // We already failed inerting this shadow root.
+      if (rootElement.__failedAttachShadow) return;
+
+      // Force shadowRoot.
+      if (!rootElement.shadowRoot) {
```

Comment:
Can we switch the order of the if and else cases here, so we're checking a positive condition instead of a negative one?